docs/source/distribution_implementation.md

```diff
@@ -0,0 +1,452 @@
+    rv_op = blah
+
+    # dist() is responsible for returning an instance of the rv_op. We pass
+    # the standard parametrizations to super().dist
+    @classmethod
+    def dist(cls, param1, param2=None, alt_param2=None, **kwargs):
+        param1 = at.as_tensor_variable(intX(param1))
+        if param2 is not None and alt_param2 is not None:
+            raise ValueError('Only one of param2 and alt_param2 is allowed')
+        if alt_param2 is not None:
+            param2 = 1 / alt_param2
+        param2 = at.as_tensor_variable(floatX(param2))
+
+        # The first value-only argument should be a list of the parameters that
+        # the rv_op needs in order to be instantiated
+        return super().dist([param1, param2], **kwargs)
+
+    # Logp returns a symbolic expression for the logp evaluation of the variable
+    # given the `value` of the variable and the parameters `param1` ... `paramN`
+    def logp(value, param1, param2):
+        logp_expression = value * (param1 + at.log(param2))
+
+         # We use `bound` for parameter validation. After the default expression,
+         # multiple comma-separated symbolic conditions can be added. Whenever
+         # a bound is invalidated, the returned expression evaluates to `-np.inf`
+         return bound(
+            logp_expression,
+            value >= 0,
+            param2 >= 0,
+            # There is one sneaky optional keyowrd argument, that converts an
+            # otherwise elemwise `bound` to a reduced scalar bound. This is usually
+            # needed for multivariate distributions where the dimensionality
+            # of the bound conditions does not match that of the "value" / "logp"
+            # By default it is set to `True`.
+            broadcast_conditions=True,
+        )
+
+    # logcdf works the same way as logp. For bounded variables, it is expected
+    # to return `-inf` for values below the domain start and `0` for values
+    # above the domain end, but only when the parameters are valid.
+    def logcdf(value, param1, param2):
+        ...
+
+```
+
+Some notes:
+
+1. A distribution should at the very least inherit from `Discrete` or
+   `Continuous`. For the latter, more specific subclasses exist:
+   `PositiveContinuous`, `UnitContinuous`, `BoundedContinuous`,
+   `CircularContinuous`, which specify default transformations for the variables.
+   If you need to specify a one-time custom transform you can also override the
+   `__new__` method, as is done for the [`Dirichlet`](https://github.com/pymc-devs/pymc3/blob/6eaf9a0d7bc5b9168bedc9337808391ccb86a5ea/pymc3/distributions/multivariate.py#L399-L401).
+1. If a distribution does not have a corresponding `random` implementation, a
+   `RandomVariable` should still be created that raises a `NotImplementedError`.
+   This is the case for the [`Flat` distribution](https://github.com/pymc-devs/pymc3/blob/6eaf9a0d7bc5b9168bedc9337808391ccb86a5ea/pymc3/distributions/continuous.py#L312-L321).
```

Comment:
With FlatRV the situation is similar yet a bit different, because FlatRV _in not part of the docs!_. I understand it should not be part of the user facing docs but it looks like it's useful for dev facing docs. Would it be a good idea to have a couple of orphan API pages (therefore not indexed in the API navbar section) so we can link there from the developer guide?

Reply:
Sounds reasonable